.tekton/check-generated-files-up-to-date-task.yaml

```
@@ -47,3 +47,24 @@ spec:
         echo >&2 "Some generated files are not up to date with their source ^^^. Run '$cmd' to update them."
         exit 1
       fi
+  - name: check-no-quay-io-references
+    image: registry.access.redhat.com/ubi8:latest
+    workingDir: /var/workdir/source
+    script: |
+      #!/usr/bin/env bash
+      set -euo pipefail
+      set -x
+
+      catalogs=("catalog-template.yaml" "catalog-bundle-object/rhacs-operator/catalog.json" "catalog-csv-metadata/rhacs-operator/catalog.json")
+
+      for catalog in "${catalogs[@]}"; do
+        if [[ ! -f "$catalog" ]]; then
+          echo >&2 "File $catalog does not exist."
+          continue
+        fi
+
+        if grep -q 'quay.io' "$catalog"; then
```

```suggestion
        if grep -qF 'quay.io' "$catalog"; then
```

Also, maybe drop `-q` so that we see what references there are?